puppet/zulip/files/nginx/zulip-include-frontend/uploads-internal.conf

```diff
@@ -1,12 +1,36 @@
-location /internal/uploads {
+# Handle redirects to S3
+location ~ ^/internal/s3/([^/]+)/(.*) {
+    internal;
+    set $download_url https://$1/$2;
+    proxy_set_header Host $1;
+
+    # Inherit the Content-Disposition which Django sent
+    proxy_hide_header Content-Disposition;
+    add_header Content-Disposition $upstream_http_content_disposition;
+    include /etc/nginx/zulip-include/headers;
+
+    # nginx does its own DNS resolution, which is necessary here to
+    # resolve the IP of the S3 server.  Point it at the local caching
+    # systemd resolved service.
+    resolver 127.0.0.53 valid=300s;
+    resolver_timeout 10s;
+
+    proxy_pass $download_url$is_args$args;
+    proxy_cache uploads;
+    proxy_cache_valid 200 24h;
```

Comment:
Hm -- I guess this should be the same as `s3_cache_inactive_time`, since it's the upper bound on how long the entry is valid.  I'll also see if we can remove this line, or which one wins if it's a mismatch with `s3_cache_inactive_time`.